client.go

```diff
@@ -80,8 +80,11 @@ var (
 type ReaderFunc func() (io.Reader, error)
 
 // ResponseHandlerFunc is a type of function that takes in a Response, and does something with it.
-// It only runs if the initial part of the request was successful.
-// If an error is returned, the client's retry policy will be used to determine whether to retry the whole request.
+// If an error is returned, the client's retry policy will be used to determine
+// whether to retry the whole request (including this handler).
+// NOTE: It only runs if the initial part of the request was successful.
```

Comment:
Also just a nitpick - I think it's fine to just flow these like a normal sentence in the description rather than marking them specifically as `NOTE:`.